qiskit_nature/second_q/operators/vibrational_op.py

```diff
@@ -265,11 +272,61 @@ def _validate_keys(self, keys: Collection[str]) -> None:
 
     @classmethod
     def _validate_polynomial_tensor_key(cls, keys: Collection[str]) -> None:
-        ...
+        allowed_chars = {"+", "-", "_"}
+
+        for key in keys:
+            if set(key) != allowed_chars:
+                raise QiskitNatureError(
+                    f"The key {key} is invalid. PolynomialTensor keys must be multiples of the "
+                    "`_+-` character sequence, for them to be expandable into a VibrationalOp.\n"
+                    "But this key contains other characters."
+                )
+            if len(key) % 3:
+                raise QiskitNatureError(
+                    f"The key {key} is invalid. PolynomialTensor keys must be multiples of the "
+                    "`_+-` character sequence, for them to be expandable into a VibrationalOp.\n"
+                    "But the length of this key is not a multiple of 3."
+                )
+            if key != "_+-" * (len(key) // 3):
```

Comment:
Yes I guess I could do it all in one check. It depends on how specific we want the error messages to be. :thinking: 
If you have no preference, I will reduce it to a single regex-based check tomorrow :+1:

Reply:
If the last message says it needs to be a sequence of 0/1 or more "_+-" strings I think its clear enough what is expected and you can look at the key value returned and figure it. Maybe with the regex and the match result it can indicate where the match fails too, but that's more work.